artifactory/commands/permissiontarget/template.go

```diff
@@ -0,0 +1,181 @@
+//	* repos - Mandatory for repo and releaseBundle. Has a const default value for build.
+//	* include/exclude-patterns - Optional, has a default value.
+//	* actions - Optional,includes two maps (users and groups): user/group name -> permissions array.
+func permissionSectionCallBack(iq *utils.InteractiveQuestionnaire, section string) (value string, err error) {
+	if section == utils.SaveAndExit {
+		return
+	}
+	var sectionAnswer PermissionSectionAnswer
+	if section != Build {
+		sectionAnswer.Repositories = utils.AskString(reposQuestionInfo.Msg, reposQuestionInfo.PromptPrefix, false)
+	}
+	sectionAnswer.IncludePatterns = utils.AskString(includePatternsQuestionInfo.Msg, includePatternsQuestionInfo.PromptPrefix, true)
+	sectionAnswer.ExcludePatterns = utils.AskString(excludePatternsQuestionInfo.Msg, excludePatternsQuestionInfo.PromptPrefix, true)
+	configureActions := utils.AskFromList("", configureActionsQuestionInfo.PromptPrefix+"users?"+utils.PressTabMsg, false, configureActionsQuestionInfo.Options)
+	if configureActions == Yes {
+		sectionAnswer.ActionsUsers = make(map[string]string)
+		readActionsMap("user", sectionAnswer.ActionsUsers)
+	}
+	configureActions = utils.AskFromList("", configureActionsQuestionInfo.PromptPrefix+"groups?", false, configureActionsQuestionInfo.Options)
+	if configureActions == Yes {
+		sectionAnswer.ActionsGroups = make(map[string]string)
+		readActionsMap("group", sectionAnswer.ActionsGroups)
+	}
+	iq.AnswersMap[section] = sectionAnswer
+	return
+}
+
+// We will read (user/group name, permissions) pairs until empty name is read.
+func readActionsMap(actionsType string, actionsMap map[string]string) {
+	fmt.Println("Permissions value is a comma separated list of values from read, write, annotate, delete, manage, managedXrayMeta, distribute")
+	keyPromptPrefix := "Insert " + actionsType + " name (press enter to finish) >"
+	for {
+		key := utils.AskString("", keyPromptPrefix, true)
+		if key == "" {
+			return
+		}
+		value := utils.AskString("", "Insert permission value for \""+key+"\" >", false)
+		actionsMap[key] = value
+	}
+}
+
+var questionMap = map[string]utils.QuestionInfo{
+	Name: {
+		Msg:          "",
+		PromptPrefix: "Insert the permission target name >",
+		AllowVars:    false,
+		Writer:       utils.WriteStringAnswer,
+		MapKey:       Name,
+		Callback:     nil,
+	},
+	utils.OptionalKey: {
+		Msg:          "",
+		PromptPrefix: SelectPermissionTargetSectionMsg,
+		AllowVars:    false,
+		Writer:       nil,
+		MapKey:       "",
+		Callback:     permissionSectionCallBack,
+	},
+	Repo:          utils.FreeStringQuestionInfo,
+	Build:         utils.FreeStringQuestionInfo,
+	ReleaseBundle: utils.FreeStringQuestionInfo,
+}
+
+var reposQuestionInfo = utils.QuestionInfo{
+	Msg:          "You can specify the name \"ANY\" in order to apply to all repositories, \"ANY REMOTE\" for all remote repositories and \"ANY LOCAL\" for all local repositories.\n" + utils.CommaSeparatedListMsg,
```

Comment:
...in order to apply to all repositories... --> ...to apply to all repositories...